### PR TITLE
New LowestCommonMultiple Goodie

### PR DESCRIPTION
**Pull request by arun-geek**
---
https://duck.co/ia/view/lowest_common_multiple

### CODE REVIEW

**Comment by daxtheduck** [comment]
@arun-geek Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/LowestCommonMultiple.t [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Test::More;
+use DDG::Test::Goodie;
+
+zci answer_type => "lowest_common_multiple";
+zci is_cached   => 1;
+#!/usr/bin/env perl
```
Comment (GuiltyDolphin) [inline t/LowestCommonMultiple.t]: Lets get rid of this.

File: t/LowestCommonMultiple.t [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+            operation => 'Lowest Common Multiple',
+
+            result    => 22
+
+        }
+
+    ),
+
+    # Etc...
```
Comment (GuiltyDolphin) [inline t/LowestCommonMultiple.t]: And this... And any others.

File: lib/DDG/Goodie/LowestCommonMultiple.pm [hunk 3]
```diff
@@ -0,0 +1,77 @@
+package DDG::Goodie::LowestCommonMultiple;
+
+
+use strict;
+use DDG::Goodie;
+
+zci answer_type => "greatest_common_factor";
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/LowestCommonMultiple.pm]: I think you mean `lowest_common_multiple`.

File: lib/DDG/Goodie/LowestCommonMultiple.pm [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+zci is_cached   => 1;
+triggers startend => 'lowest common multiple', 'lowest common multiple of', 'lcm','lcm of','least common multiple';
+
+handle remainder => sub {
+
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/LowestCommonMultiple.pm]: Please fix the spacing throughout the file. We don't need all these blank lines everywhere.

File: lib/DDG/Goodie/LowestCommonMultiple.pm [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        $result = lcm($result, $_)
+
+    }
+
+
+
+    return "Lowest Common Multiple of $formatted_numbers is $result.",
+
+      structured_answer => {
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/LowestCommonMultiple.pm]: We can use a template (`text`) here.

File: t/LowestCommonMultiple.t [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    [qw( DDG::Goodie::LowestCommonMultiple )],
+
+    'lcm 9 81' => test_zci(
+
+        'Lowest Common Multiple of 9 and 81 is 81.',
+
+        structured_answer => {
+
+            input     => ['9 and 81'],
+
```
Comment (GuiltyDolphin) [inline t/LowestCommonMultiple.t]: Spacing should be fixed in this file too.

File: t/LowestCommonMultiple.t [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            input     => ['11 and 2'],
+
+            operation => 'Lowest Common Multiple',
+
+            result    => 22
+
+        }
+
+    ),
```
Comment (GuiltyDolphin) [inline t/LowestCommonMultiple.t]: Add some tests for when it shouldn't trigger (bad queries, invalid numbers etc...)

**Comment by GuiltyDolphin** [comment]
The IA page should be `lowest_common_multiple`, @moollaza.

**Comment by moollaza** [comment]
>The IA page should be lowest_common_multiple, @moollaza.

IA Page has been updated :+1:

**Comment by moollaza** [comment]
@arun-geek thanks again for taking the time to create this Instant Answer ! You done a great job so far! We at DuckDuckGo, the DuckDuckHack community, and all the people who will encounter this Instant Answer will really appreciate it.

It looks like GuiltyDolphin has left you some feedback that need to be implemented. Are you still free to finish up this Instant Answer?

I'd love to chat if you're around this week or next to help you get this IA finished. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Comment by moollaza** [comment]
@arun-geek I'm going to close this for now. If you'd like to finish this up later please let us know!

**Pull request closed without merging**